burotel/indico_burotel/tasks.py

```diff
@@ -0,0 +1,188 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2014 - 2021 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+
+import requests
+from celery.exceptions import Retry
+from celery.schedules import crontab
+from flask_pluginengine import current_plugin
+from requests.exceptions import RequestException, Timeout
+from sqlalchemy.types import Interval
+
+from indico.core.auth import multipass
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.modules.rb.models.reservations import Reservation, ReservationEditLog, ReservationState
+from indico.modules.rb.models.room_attributes import RoomAttribute, RoomAttributeAssociation
+from indico.modules.rb.models.rooms import Room
+from indico.modules.users.models.users import User
+from indico.util.date_time import now_utc
+
+from indico_burotel.notifications import notify_about_to_cancel, notify_automatic_cancellation
+
+
+def _get_cern_ident(plugin, user):
+    cern_ident_provider = plugin.settings.get('cern_identity_provider')
+    cern_ident = next(
+        (ident for ident in user.identities if ident.provider == cern_ident_provider),
+        None
+    )
+    if not cern_ident or not cern_ident.multipass_data or not cern_ident.multipass_data.get('cern_person_id'):
+        return None
+    else:
+        return cern_ident
+
+
+def _find_person_id(plugin, user):
+    """Get the CERN Person ID of a user."""
+    cern_ident_provider = plugin.settings.get('cern_identity_provider')
+    cern_ident = _get_cern_ident(plugin, user)
+
+    if not cern_ident:
+        # try to fetch the identity from the source before giving up
+        results = multipass.search_identities(providers={cern_ident_provider}, exact=True, email=user.all_emails)
+        if not (cern_ident := next(results, None)):
+            # that didn't work out, raise an error
+            raise Exception("Can't find %s identity for %s", cern_ident_provider, user)
```

Comment:
```suggestion
            raise Exception(f"Can't find {cern_ident_provider} identity for {user}")
```

only loggers take format args like this